src/test/ui/consts/validate_never_arrays.stderr

```diff
@@ -21,6 +27,9 @@ LL | const _: &[!] = unsafe { &*(1_usize as *const [!; 42]) };
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ type validation failed: encountered a value of the never type `!` at .<deref>[0]
```

Comment:
The actual allocation isn't useful for this error.

Reply:
it may be useful, if the user expected to have a different enum variant which would actually be inhabited but due to some flipped bits in the discriminant got the uninhabited variant.

Reply:
I'm willing to go either way, but let's make a decision here? (This conversation isn't resolved, so asking just to be sure).

Reply:
I'd say let's leave it in for now and look at more cases as users start seeing this